modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/DeviceUtil.java

```diff
@@ -52,9 +52,10 @@ public static void persistChallengeToken(String token, String deviceID, String u
         } else {
             cacheEntry = (CacheEntry) objCacheEntry;
         }
-        cacheEntry.setDeviceID(deviceID);
+        if (deviceID != null) {
+            cacheEntry.setDeviceID(deviceID);
```

Comment:
device Id is necessary to pass.Even though we can not take it from primary responses.